app/role.go

```diff
@@ -0,0 +1,128 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/temporalio/tcld/protogen/api/auth/v1"
+	"github.com/temporalio/tcld/protogen/api/authservice/v1"
+)
+
+var (
+	accountActionGroups   = getAccountActionGroups()
+	namespaceActionGroups = getNamespaceActionGroups()
+)
+
+func getAccountActionGroups() []string {
+	var rv []string
+	for n, v := range auth.AccountActionGroup_value {
+		if v != int32(auth.ACCOUNT_ACTION_GROUP_UNSPECIFIED) {
+			rv = append(rv, n)
+		}
+	}
+	return rv
+}
+
+func getNamespaceActionGroups() []string {
+	var rv []string
+	for n, v := range auth.NamespaceActionGroup_value {
+		if v != int32(auth.NAMESPACE_ACTION_GROUP_UNSPECIFIED) {
+			rv = append(rv, n)
+		}
+	}
+	return rv
+}
+
+func getNamespaceRolesFromMap(ctx context.Context, client authservice.AuthServiceClient, namespaceActionGroups map[string]string) ([]*auth.Role, error) {
+	var res []*auth.Role
+	for namespace, actionGroup := range namespaceActionGroups {
+		r, err := getNamespaceRole(ctx, client, namespace, actionGroup)
+		if err != nil {
+			return nil, err
```

Comment:
[Nit] is a multierr a better user experience here?

Reply:
changed to use batch api so not relevant now.